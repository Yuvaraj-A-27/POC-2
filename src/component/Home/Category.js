import { makeStyles, Paper, Typography } from '@material-ui/core'
import axios from 'axios'
import React, { useEffect, useState } from 'react'

const useStyles = makeStyles((theme)=>({
    heading:{
        textAlign:'left',
        marginTop:'3%',
        marginLeft : '3%',
        marginRight : '3%',
        fontWeight:'800',
        background: 'linear-gradient(10deg, #9269fa 40%, #d9d9d9 100%)',
        border: 0,
        fontSize: 20,
        borderRadius: 3,
        color: 'white',
        height: 25,
        padding: '10px 30px',


    },
    hr:{
        marginTop:'1%',
        margin: theme.spacing(5)
    },
    paperDiv:{
        display : 'flex',
        flexWrap : 'wrap',
        '& > *': {
            marginLeft: theme.spacing(10),
            marginTop: theme.spacing(1),
            width: theme.spacing(25),
            height: theme.spacing(7),
        }
    },
    paper:{
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
        fontSize : 28,
        padding: '2px 20px 0px 20px',
        color: 'white',
        height : '10%',
    }
}))

function Category(){
    const classes = useStyles()

    const [category, setCategory] = useState()

    useEffect(()=>{
        axios.get("https://fakestoreapi.com/products/categories")
        .then(res => setCategory(res.data))
        .catch(err =>console.log(err))
    },[])

    const categories = category.map((e,index)=>(
        <Paper className={classes.paper} elevation={3} key={index}>{e}</Paper>
    ))
    return(
        <div>
            <Typography className={classes.heading} variant ='h5'>Category</Typography>
            <hr className={classes.hr}/>
            <div className={classes.paperDiv}>
                {categories}
            </div>
        </div>
    )
}

export default Category