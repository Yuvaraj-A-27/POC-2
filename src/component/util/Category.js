import { makeStyles, Paper, Typography } from '@material-ui/core'
import React from 'react'
import { useSelector } from 'react-redux'

const useStyles = makeStyles((theme)=>({
    heading:{
        textAlign:'left',
        marginTop:'3%',
        marginLeft : '3%',
        fontWeight:'800',
        width: '30%',
        background: 'linear-gradient(10deg, #2c0ccc 5%, #7665c9 85%)',
        border: 0,
        fontSize: 20,
        borderRadius: 15,
        color: 'white',
        height: 25,
        padding: '10px 30px',
        transition : 'width 1s',
        '&:hover':{
            width: '80%'
        }
    },
    hr:{
        marginTop:'1%',
        marginLeft: theme.spacing(5),
        marginRight: theme.spacing(5),
        marginBottom: theme.spacing(10),

    },
    paperDiv:{
        display : 'flex',
        flexWrap : 'wrap',
        '& > *': {
            marginLeft: theme.spacing(10),
            marginTop: theme.spacing(1),
            marginBottom: theme.spacing(3),
            width: theme.spacing(25), 
        },
        padding: '50px 0px 1px',
        
    },
    paper:{
        background: 'linear-gradient(45deg, #FE6B8B 30%, #FF8E53 90%)',
        fontSize : 28,
        padding: '2px 20px 0px 20px',
        color: 'white',
        height : '10%',
        transition : 'transform 0.5s',
        '&:hover':{
            transform: 'translateY(-10px)'
        }
    },
    aHref:{
        textDecoration : 'none',
        color: 'white',
    }
}))

function Category(props){
    const classes = useStyles()

    let category = useSelector(state => state.category)
    let categories = null
    if(category){
        categories = category.map((e,index)=>(
            // <Paper className={classes.paper} elevation={3} key={index}>{e}</Paper>
            <Paper className={classes.paper} elevation={3} key={index}><a className={classes.aHref} href={'#'+e}>{e}</a></Paper>
        ))
    }
    return(
        <div data-testid={props.dataTestid}>
            <Typography className={classes.heading} variant ='h5'>Category</Typography>
            <div className={classes.paperDiv}>
                {categories}
            </div>
            <hr className={classes.hr}/>
        </div>
    )
}

export default Category