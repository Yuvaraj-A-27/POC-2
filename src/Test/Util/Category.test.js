import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
// import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import Category from '../../component/util/Category'


const middleware  = {}
const mockStore = configureStore(middleware)
const initialState = {}
const Store = mockStore(initialState)


test('Category Snapshot', ()=>{
    const category = renderer.create(
        <Provider store={Store}>
            <Category />
        </Provider>
    ).toJSON()
    expect(category).toMatchSnapshot()
})