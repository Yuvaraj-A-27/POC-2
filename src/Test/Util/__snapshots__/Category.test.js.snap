// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Snapshot 1`] = `
<div>
  <h5
    className="MuiTypography-root makeStyles-heading-1 MuiTypography-h5"
  >
    Category
  </h5>
  <div
    className="makeStyles-paperDiv-3"
  />
  <hr
    className="makeStyles-hr-2"
  />
</div>
`;
